lib/db/leveldb_dbinstance.go

```diff
@@ -351,12 +351,6 @@ func (db *Instance) withNeed(folder, device []byte, truncate bool, needAllInvali
 			if bytes.Equal(v.Device, device) {
 				have = true
 				haveFileVersion = v
-				// We need invalid files regardless of version when
```

Comment:
What's the thing behind the need invalid stuff that we just added now no longer making sense? Just a couple of sentences of background for my edification?

(It's a minus diff and I was confused when it was added so I'm not complaining, it's just that I'm now confused again.)

Reply:
Before the invalidation of ignored files was introduced, `WithNeed` always returned ignored files, as it's version wasn't updated. Their version was updated to keep "in-sync" with the global version. Therefore these files would not get pulled unless the global version gets updated. The removed option returned all invalid files and was triggered when ignore patterns change. With the new mechanism of setting the version to conflict with the global state when unignoring, this is no longer needed.